hooks/submit_for_review.py

```diff
@@ -263,56 +268,57 @@ def publish(self, settings, item):
         :param item: Item to process
         """
 
-        sg_publish = item.properties["sg_publish_data"]
+        render_path = item.properties.get("path")
+        if render_path is None:
+            raise Exception("'path' property cannot be found in item's properties. "
+                            "Review Submission failed.")
+
+        sg_publish_data = item.properties.get("sg_publish_data")
+        if sg_publish_data is None:
+            raise Exception("'sg_publish_data' was not found in the item's properties. "
+                            "Review Submission for '%s' failed." % render_path)
         sg_task = self.parent.context.task
         comment = item.description
         thumbnail_path = item.get_thumbnail_as_path()
         progress_cb = lambda *args, **kwargs: None
         review_submission_app = self.parent.engine.apps.get("tk-multi-reviewsubmission")
 
-        render_path = item.properties.get("path")
         render_template = item.properties.get("work_template")
+        if render_template is None:
+            raise Exception("'work_template' property is missing from item's properties. "
+                            "Review submission for '%s' failed." % render_path)
         publish_template = item.properties.get("publish_template")
+        if publish_template is None:
+            raise Exception("'publish_template' property not found on item. "
+                            "Review submission for '%' failed." % render_path)
+        if not render_template.validate(render_path):
+            raise Exception("'%s' did not match the render template. "
+                            "Review submission failed." % render_path)
+
         render_path_fields = render_template.get_fields(render_path)
         first_frame = item.properties.get("first_frame")
         last_frame = item.properties.get("last_frame")
-
-        if hasattr(review_submission_app, "render_and_submit_version"):
-            # this is a recent version of the review submission app that contains
-            # the new method that also accepts a colorspace argument.
-            colorspace = item.properties.get("color_space")
-            review_submission_app.render_and_submit_version(
+        colorspace = item.properties.get("color_space")
+        
+        if review_submission_app.render_and_submit_version(
```

Comment:
I'd suggest storing the Version entity dictionary returned and showing an action button that allows the user to jump directly to it in SG. This would be similar to this: https://github.com/shotgunsoftware/tk-multi-publish2/blob/45377_context_change/hooks/upload_version.py#L303-L312

Reply:
fixed